packages/@aws-cdk/aws-codebuild/lib/source.ts

```diff
@@ -261,7 +251,6 @@ export class GitHubSource extends GitBuildSource {
 
   protected toSourceProperty(): any {
     return {
-      auth: { type: 'OAUTH', resource: this.oauthToken },
```

Comment:
I don't think this is correct. I believe you still need to supply `auth: { type: 'OAUTH' }` here.

Reply:
@Kaixiang-AWS can you confirm whether the `auth: { type: 'OAUTH' }` is still required here, even without the Token?

Reply:
I have updated the PR and added an integ test against github source to make sure it can deploy.

I can confirm that `auth` field is not required. There might be an issue with the doc. I will report it to the doc writer and ask him to update the doc.